File: src/universe/universe.cpp
// universe.cpp
#include "universe.h"
#include <random>

Universe::Universe(int size, int numCreatures)
    : size(size), numCreatures(numCreatures), cells(std::make_unique<std::vector<bool>>()) {
    initializeUniverse();
}

const std::vector<bool>& Universe::getCells() const {
    return *cells;
}

void Universe::initializeUniverse() {
    cells->resize(size, false);
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<int> distribution(0, size - 1);

    for (int i = 0; i < numCreatures; ++i) {
        int idx = distribution(gen);
        (*cells)[idx] = true;
    }
}

File: universe.py
# universe.py

import random

class Universe:
    def __init__(self, size, num_creatures):
        self.size = size
        self.num_creatures = num_creatures
        self.cells = self.initialize_universe()

    def initialize_universe(self):
        universe = [False] * self.size
        alive_indices = random.sample(range(self.size), self.num_creatures)
        for idx in alive_indices:
            universe[idx] = True
        return universe
